### ISSUE TITLE
jdbi not support non alpahbet charaset in sql

### ISSUE THREAD

**Issue by ppsnumi** [jdbi/jdbi#1487]
Hi~

I meet some error using jdbi3(v3.6.0)
that's ORA-00936 error.

```
sql : SELECT nonAlphabetCharacterColumnName FROM nonAlphabetCharacter_TableName
rewritten: "SELECT FROM _"
```
oracle throw error.
database can using locale encoding database object name.

how to resolve this error???

**Comment by qualidafial** [comment]
Can you give an example of the characters that are being stripped out of the rewritten SQL?

**Comment by ppsnumi** [comment]
"SELECT 제목 FROM 업무_게시물"

remove all Non-alphabet character in jdbi when execute query.

"SELECT FROM _"

**Comment by qualidafial** [comment]
Are you configuring a specific template engine (e.g. `setTemplateEngine()`/`@UseTemplateEngine`)?

**Comment by ppsnumi** [comment]
does not use templateEngine and encoding is EUC-KR(Korean)

**Comment by qualidafial** [comment]
Okay, I'm able to reproduce it, and have a unit test, but I may not be able to release a fix right away.

As a temporary workaround (assuming you're not using `define()` or `@Define` in your application), I suggest calling `Jdbi.setTemplateEngine(TemplateEngine.NOP)` to bypass the built-in template engine.

**Comment by ppsnumi** [comment]
Ok. I'm try it.
but throw other exception.

use:
Jdbi.setTemplateEngine(TemplateEngine.NOP);

exception:
throw UnableToCreateStatementException: Exception parsing for named parameter replacement ...
stack trace ...
cause by: java.lang.IllegalArgumentException: NoViableAltException('제'@[])
stack trace ...
cause by: NoViableAltException('제'@[])
at org.jdbi.v3.core.internal.lexer.ColonStatementLexer.mTokens(ColonStatementLexer.java:1242)
at org.antlr.runtime.Lexer.nextToken(Lexer.java:89)
... 65 more

**Comment by stevenschlansker** [comment]
It’s possible double quoting the Korean identifiers could help the parser avoid that exception.

**Comment by ppsnumi** [comment]
EXE1:
use:
 Jdbi.setTemplateEngine(TemplateEngine.NOP);

sql:
"SELECT \"제목\" FROM \"업무_게시물\""

no exception and work fine.

EXE2:
I try to use alphabet alias name to euc-kr column name for bean mapping.

use:
 Jdbi.setTemplateEngine(TemplateEngine.NOP);

sql:
"SELECT \"제목\" as title FROM \"업무_게시물\""

use mapToBean(SomeBean.class) method for bean mapping(has title field).
no exception and work fine.
 
thanks.

**Comment by qualidafial** [comment]
If you're enquoting your object names you probably don't need to bother with the `TemplateEngine.NOP` thing.

**Comment by qualidafial** [comment]
This should now be fixed as part of 3.7.0, just released to Maven central.

In terms of internationalization support, 3.7.0:

* Leaves Unicode literals intact in SQL. We used your examples above in our tests.
* Allows any valid Java identifier (not just English alphabet) as a named parameter in SQL statements, e.g. `WHERE title = :제목`.
* Allows any valid Java identifier as a defined attribute in SQL statement, e.g. `SELECT <제목> FROM book`
